source/acquaman/AMScanActionControllerBasicFileWriter.cpp

```diff
@@ -62,6 +65,12 @@ AMScanActionControllerBasicFileWriter::AMScanActionControllerBasicFileWriter(con
 		if(!spectraFile_->open(QIODevice::WriteOnly | QIODevice::Text))
 			emitError(AMScanActionControllerBasicFileWriter::CouldNotOpenError);
 	}
+
+	// start the timer (2mins) to check whether we are writing to the files consistently
+	QTimer *dataFileSizeCheckTimer = new QTimer(this);
```

Comment:
I just checked to make sure, the file writer is also explicitly destroyed by calling deleteLater() some where on it. That was the file writer fix during the shutdown.